lib/matplotlib/text.py

```diff
@@ -759,9 +759,12 @@ def draw(self, renderer):
             # position in Text, and dash position in TextWithDash:
             posx = float(textobj.convert_xunits(textobj._x))
             posy = float(textobj.convert_yunits(textobj._y))
-            if not np.isfinite(posx) or not np.isfinite(posy):
-                raise ValueError("posx and posy should be finite values")
             posx, posy = trans.transform_point((posx, posy))
+            if not np.isfinite(posx) or not np.isfinite(posy):
+                warnst = "x and y are not finite values for text "
+                warnst += "string '%s'.  Not rendering text."
+                warnings.warn(warnst%self.get_text())
```

Comment:
Please use str.format and implicit string concat, something like
```
warnings.warn("x and y ..."
              "...".format(self.get_text()))
```
Otherwise looks good.

Reply:
Since it used to throw an exception before, I can't imagine too many people are going to complain if it throws a warning.